ICAFIX/scripts/computeRecleanFeatures.m

```diff
@@ -0,0 +1,270 @@
+        CIFTIMSMAllSmoothVN=ciftiopen([SubjFolderlist '/MNINonLinear/Results/' fMRIName '/' fMRIName '_Atlas_MSMAll_hp' hp '_clean_vn_' num2str(FinalSpatialSmoothingFWHM) '.dscalar.nii'],'wb_command');
+        CIFTIMSMAllSmooth.cdata=CIFTIMSMAllSmooth.cdata./repmat(CIFTIMSMAllSmoothVN.cdata,1,size(CIFTIMSMAllSmooth.cdata,2));           
+        %CIFTIDropouts=ciftiopen([SubjFolderlist '/MNINonLinear/Results/' fMRIName '/' fMRIName '_dropouts.dscalar.nii'],'wb_command');
+        cifti_dropout_file=[SubjFolderlist '/MNINonLinear/Results/' fMRIName '/' fMRIName '_dropouts.dscalar.nii'];
+
+        CIFTIDropouts=ciftiopen(cifti_dropout_file,'wb_command');
+        WMPARC=read_avw([SubjFolderlist '/MNINonLinear/ROIs/wmparc.' Resolution '.nii.gz']);
+        ROIFolder=[SubjFolderlist '/MNINonLinear/ROIs'];
+
+        system(['wb_command -volume-label-import ' ROIFolder '/wmparc.' Resolution '.nii.gz ' CSFLabelFile ' ' ROIFolder '/CSFReg.' Resolution '.nii.gz -discard-others -drop-unused-labels']);
+        system(['wb_command -volume-label-import ' ROIFolder '/wmparc.' Resolution '.nii.gz ' WMLabelFile ' ' ROIFolder '/WMReg.' Resolution '.nii.gz -discard-others -drop-unused-labels']);
+
+        WM=read_avw([SubjFolderlist '/MNINonLinear/ROIs/WMReg.' Resolution '.nii.gz']);
+        CSF=read_avw([SubjFolderlist '/MNINonLinear/ROIs/CSFReg.' Resolution '.nii.gz']);
+        % dropout file is not ready for lifespan
+        %DROPOUT=read_avw([SubjFolderlist '/MNINonLinear/Results/' fMRIName '/' fMRIName '_dropouts.nii.gz']);
+        dropout_file=[SubjFolderlist '/MNINonLinear/Results/' fMRIName '/' fMRIName '_dropouts.nii.gz'];
+
+        DROPOUT=read_avw(dropout_file);
+        VOLUME=read_avw([SubjFolderlist '/MNINonLinear/Results/' fMRIName '/' fMRIName '_hp' hp '.ica/filtered_func_data.ica/melodic_oIC.nii.gz']);
+        VOLUME=reshape(VOLUME,size(VOLUME,1)*size(VOLUME,2)*size(VOLUME,3),size(VOLUME,4));
+        Volume_data=VOLUME(std(VOLUME,[],2)>0,:);
+        WMPARC=reshape(WMPARC,size(WMPARC,1)*size(WMPARC,2)*size(WMPARC,3),size(WMPARC,4));
+        wmparc_data=WMPARC(std(VOLUME,[],2)>0,:);
+        WM=reshape(WM,size(WM,1)*size(WM,2)*size(WM,3),size(WM,4));
+        wm_data=WM(std(VOLUME,[],2)>0,:);
+        wm_data(wm_data~=0)=1;
+        CSF=reshape(CSF,size(CSF,1)*size(CSF,2)*size(CSF,3),size(CSF,4));
+        csf_data=CSF(std(VOLUME,[],2)>0,:);
+        csf_data(csf_data~=0)=1;
+        DROPOUT=reshape(DROPOUT,size(DROPOUT,1)*size(DROPOUT,2)*size(DROPOUT,3),size(DROPOUT,4));
+        dropout_data=DROPOUT(std(VOLUME,[],2)>0,:);
+        SBREF=read_avw([SubjFolderlist '/MNINonLinear/Results/' fMRIName '/' fMRIName '_SBRef.nii.gz']);
+        SBREFOrig=SBREF;
+        SBREF=reshape(SBREF,size(SBREF,1)*size(SBREF,2)*size(SBREF,3),size(SBREF,4));
+        sbref_data=SBREF(std(VOLUME,[],2)>0,:);
```

Comment:
```suggestion
        SBREFOrig = read_avw([SubjFolderlist '/MNINonLinear/Results/' fMRIName '/' fMRIName '_SBRef.nii.gz']);
        
        function [maskeddata, flatdata] = vol_reshape_and_mask(indata, mask)
            flatdata = reshape(indata, prod(size(indata, [1 2 3])), size(indata, 4));
            maskeddata = flatdata(mask, :);
        end
        
        VOLUME = reshape(VOLUME, prod(size(VOLUME, [1 2 3])), size(VOLUME, 4));
        volumeValid = range(VOLUME, 2) > 0;
        Volume_data = VOLUME(volumeValid, :);
        
        [wmparc_data, WMPARC] = vol_reshape_and_mask(WMPARC, volumeValid);
        wm_data = vol_reshape_and_mask(WM, volumeValid);
        csf_data = vol_reshape_and_mask(CSF, volumeValid);
        dropout_data = vol_reshape_and_mask(DROPOUT, volumeValid);
        sbref_data = vol_reshape_and_mask(SBREFOrig, volumeValid);
        
        wm_data(wm_data ~= 0) = 1;
        csf_data(csf_data ~= 0) = 1;
```
Please don't mix other operations into the middle of a block of otherwise repetitive things.  I'm guessing this was copy and paste from other code, though, so probably not your fault.  I didn't keep most of the flattened but unmasked versions, as I think they are unused, but WMPARC gets used as a template to reconstruct back into full-volume.

`range` is simpler, faster, and more stable than `std` for this purpose, and recomputing the same ROI over and over is silly and less obvious/readable.  Please also replace the other `std(VOLUME, [], 2) > 0` stuff below with `volumeValid`.

I believe this function syntax is valid for nested functions, and has been for many matlab versions (though it doesn't need any of the special weirdness of nested functions, this was just an obvious place to put it in the code, for readability).  If it doesn't work, or looks weird to people, it could be moved closer to the end of the file, becoming a local function, like myreadtext.